src/build_interpret.ml

```diff
@@ -42,9 +42,13 @@ let static_deps t ~all_targets_by_dir =
         match !state with
         | G_evaluated l ->
           { acc with action_deps = Pset.union acc.action_deps (Pset.of_list l) }
-        | G_unevaluated (dir, re) ->
+        | G_unevaluated (loc, dir, re) ->
           match Pmap.find dir (Lazy.force all_targets_by_dir) with
-          | None -> acc
+          | None ->
+            Loc.warn loc "Directory %s doesn't exist."
+              (Path.to_string_maybe_quoted dir);
+            state := G_evaluated [];
```

Comment:
Did this mean that the non-existent glob would be executed more than once? So it's a bit more than just better error reporting?

Reply:
It's not documented, but `static_deps` is assumed to turn all the `Paths_glob` from `G_unevaluated` to `G_evaluated`. It wasn't done in this branch and it was a mistake